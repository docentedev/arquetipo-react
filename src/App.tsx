import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import {  useManagerInit, useStateManager }Â from 'use-state-manager';

import { textManager, yearManager} from './store/managers';

import './App.css';

import routes from './config/routes';

import Home from './containers/home/Home';

/**
 * @author Claudio Rojas <claudio.dcv@gmail.com>
 * @since v1.0.0
 * @version v1.0.0
 * @param props
 */
const App: React.FC = () => {
  
  useManagerInit(textManager);
  useManagerInit(yearManager);

  const [year, setYear] = useStateManager(yearManager);

  return (
    <div className="app">
      {year}
      <input value={year} onChange={({ target }) => setYear(target.value)} />
      <Router>
        <Route path={routes.home} component={Home} />
      </Router>
    </div>
  );
}

export default App;
