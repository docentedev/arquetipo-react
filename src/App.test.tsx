import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import App from './App';

Enzyme.configure({ adapter: new Adapter() });

describe('test to App', () => {
  test('App is ok renderer', () => {
    const component = renderer.create(
        <App />,
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
  });
  
  test('App input change', () => {
    const event = { target: { name: 'pollName', value: 'spam' } };
    const button = shallow((<App />));
    button.find('input').simulate('change', event);
  
  });
});