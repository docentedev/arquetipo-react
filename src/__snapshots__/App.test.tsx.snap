// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App is ok renderer 1`] = `
<div
  className="app"
>
  1990
  <input
    onChange={[Function]}
    value={1990}
  />
  <div
    className="home"
  >
    1990
    <button
      onClick={[Function]}
    >
      Cambiar año
    </button>
    <div
      className="date-display"
    >
      12/19/2019
    </div>
    <div
      className="card-row"
    />
  </div>
</div>
`;

exports[`test to App App is ok renderer 1`] = `
<div
  className="app"
>
  1990
  <input
    onChange={[Function]}
    value={1990}
  />
  <div
    className="home"
  >
    1990
    <button
      onClick={[Function]}
    >
      Cambiar año
    </button>
    <div
      className="date-display"
    >
      12/19/2019
    </div>
    <div
      className="card-row"
    />
  </div>
</div>
`;

exports[`test to App Home is ok renderer 1`] = `
<div
  className="app"
>
  1990
  <input
    onChange={[Function]}
    value={1990}
  />
  <div
    className="home"
  >
    1990
    <button
      onClick={[Function]}
    >
      Cambiar año
    </button>
    <div
      className="date-display"
    >
      12/19/2019
    </div>
    <div
      className="card-row"
    />
  </div>
</div>
`;
